web/src/views_util.js

```diff
@@ -36,7 +36,9 @@ export function show(opts) {
 
     unread_ui.hide_unread_banner();
     opts.update_compose();
-    narrow_state.reset_current_filter();
+    if (opts.reset_current_filter) {
+        narrow_state.reset_current_filter();
+    }
```

Comment:
Can you explain what this is doing? I would have thought that `narrow_state.reset_current_filter` would be harmless when visiting a stream-recent view; at least I don't see you setting `narrow_state.current_filter`?

Reply:
I think I did this because it was introduced for the inbox view as I was working on this PR and I didn't think I needed it, but looking now I don't see any reason not to just always reset the filter here.